File: frontend/src/app/models/ZahtjevSearch.ts
export interface ZahtjevSearch {
  userIdMin?: number|null;
  userIdMax?: number|null;
  imeContains?: string|null;
  prezimeContains?: string|null;

  knjigaNazivContains?: string|null;
  knjigaKategorijaContains?: string|null;
  knjigaOpisContains?: string|null;
  knjigaGodinaIzdavanjaMin?: number|null;
  knjigaGodinaIzdavanjaMax?: number|null;
  knjigaAutorContains?: string|null;
  knjigaKolicinaMin?: number|null;
  knjigaKolicinaMax?: number|null;
}

File: biblioteka-service/src/main/java/com/projekat/biblioteka_service/DTO/ZahtjevSearch.java
package com.projekat.biblioteka_service.DTO;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ZahtjevSearch {
    private Long userIdMin;
    private Long userIdMax;
    private String imeContains;
    private String prezimeContains;

    private String knjigaNazivContains;
    private String knjigaKategorijaContains;
    private String knjigaOpisContains;
    private Integer knjigaGodinaIzdavanjaMin;
    private Integer knjigaGodinaIzdavanjaMax;
    private String knjigaAutorContains;
    private Integer knjigaKolicinaMin;
    private Integer knjigaKolicinaMax;
}
